packages/core/src/components/essentials/threshold.ts

```diff
@@ -0,0 +1,166 @@
+			thresholdLine.attr("y2", yScaleEnd);
+			thresholdRect.attr("width", yScaleEnd - yScaleStart);
+			thresholdRect.classed("rotate", true);
+		}
+
+		const self = this;
+		this.services.events.addEventListener(Events.Threshold.SHOW, e => {
+			const hovered = e.detail.hoveredElement.node();
+			if (hovered === self.threshold) {
+				this.setThresholdLabelPosition();
+				this.label.classed("hidden", false);
+			}
+		});
+
+		this.services.events.addEventListener(Events.Threshold.HIDE, e => {
+			this.label.classed("hidden", true);
+		});
+
+		this.appendThresholdLabel();
+
+		this.addEventListeners();
+	}
+
+	getFormattedValue() {
+		const { value, axisPosition } = this.configs;
+		const options = Tools.getProperty(this.model.getOptions());
+		const scaleType = this.services.cartesianScales.getScaleTypeByPosition(axisPosition);
+
+		if (scaleType === ScaleTypes.TIME) {
+			const isVertical = [AxisPositions.LEFT, AxisPositions.RIGHT].includes(axisPosition);
+			const mainXScale = this.services.cartesianScales.getMainXScale();
+			const mainYScale = this.services.cartesianScales.getMainYScale();
+			const scale = isVertical ? mainYScale : mainXScale;
+
+			const timeScaleOptions = Tools.getProperty(options, "timeScale");
+			const timeInterval = computeTimeIntervalName(scale.ticks());
+			return formatTick(value, 0, timeInterval, timeScaleOptions);
+		}
+
+		return value.toLocaleString("en");
+	}
+
+	appendThresholdLabel() {
+		const { value, valueFormatter, fillColor, label = "Threshold" } = this.configs;
+		const holder = select(this.services.domUtils.getHolder());
+		const formattedValue = valueFormatter ? valueFormatter(value) : this.getFormattedValue();
+
+		this.label = DOMUtils.appendOrSelect(holder, `div.${this.thresholdClass}--label.${this.thresholdIdentifierClass}`);
+		this.label.html(`${label}: ${formattedValue}`);
+		this.label.classed("hidden", true);
+		this.label.style("background-color", fillColor);
+	}
+
+	setThresholdLabelPosition() {
+		const holder = this.services.domUtils.getHolder();
+		const target = this.label.node();
+		const mouseRelativePos = mouse(holder);
+
+		// Find out whether threshold label should be shown on the left or right side
+		const bestPlacementOption = this.positionService.findBestPlacementAt(
+			{
+				left: mouseRelativePos[0],
+				top: mouseRelativePos[1]
+			},
+			target,
+			[
+				PLACEMENTS.RIGHT,
+				PLACEMENTS.LEFT,
+				PLACEMENTS.TOP,
+				PLACEMENTS.BOTTOM
+			],
+			() => ({
+				width: holder.offsetWidth,
+				height: holder.offsetHeight
+			})
+		);
+
+		// Get coordinates to where label should be positioned
+		const pos = this.positionService.findPositionAt(
+			{
+				left: mouseRelativePos[0],
+				top: mouseRelativePos[1]
+			},
+			target,
+			bestPlacementOption
+		);
+
+		this.positionService.setElement(target, pos);
+	}
+
+	addEventListeners() {
+		const self = this;
+
+		DOMUtils.appendOrSelect(this.threshold, "rect")
+			.on("mouseover", function() {
```

Comment:
```suggestion
			.on("mouseover mousemove", function() {
```
since the threshold is using the mouse position to determine it's location, I'm wondering if we would want it to follow the mouse until out of the threshold zone? It definitely seems smoother IMO

Reply:
Thank you, it works better like this